src/meta/src/manager/catalog/fragment.rs

```diff
@@ -430,14 +451,89 @@ where
         bail!("fragment not found: {}", fragment_id)
     }
 
+    pub async fn pre_apply_reschedules(
```

Comment:
Better to add some docs for this function to describe the main behavior.

Reply:
I think we should not update the `vnode_mapping` before scheduling the scale commands. Consider that our meta service crashes and recovers right after we do this, then the vnode mapping of the to-be-scaled fragment will be incorrect, and the frontend will fail to schedule the `SeqScan` eventually.

- Pushing new actors and marking them `Building` will be okay just like what we also do in creating materialized views, as we ignore these not-running actors in most cases.
- Updating the `vnode_mapping` in the meta store should be done in `post_collect`, and this should also send a notification of updating with `NotificationManager`.

Reply:
You can use the `notify_fragment_mapping` function to notify fragment mapping to frontend in `FragmentManager` after #5075 merged.